src/hooks/useDynamicColorGroup.tsx

```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import { generateColors } from '../components/BrandConfigProvider/theme/generateColors';
+
+interface ColorGroup {
+  color?: string;
+  onColor?: string;
+  colorContainer?: string;
+  onColorContainer?: string;
+  backdrop?: string;
+}
+
+export const useDynamicColorGroup = (color: string) => {
+  const [colorGroup, setColorGroup] = React.useState<ColorGroup>({});
+
+  React.useEffect(() => {
+    const colorPalette = generateColors(color);
+
+    setColorGroup({
+      color: colorPalette.primary,
+      onColor: colorPalette.onPrimary,
+      colorContainer: colorPalette.primaryContainer,
+      onColorContainer: colorPalette.onPrimaryContainer,
+      backdrop: colorPalette.backdrop,
+    });
+  }, [color]);
```

Comment:
It would probably be better to use a `useMemo` here unless `generateColors` is an expensive operation. I think this will cause an extra render

Reply:
Yep, you are correct.  The `useMemo` allowed me to remove the `useState` hook, saving an extra render.